src/document.rs

```diff
@@ -297,6 +297,20 @@ impl Document {
         Ok(html.to_owned())
     }
 
+    /// Extracts references from markdown content.
+    pub fn extract_references(&self) -> String {
+        let mut trail = String::new();
+        let re = Regex::new(r"(?m:^ {0,3}\[[^\]]+\]:.+$)").unwrap();
+
+        if re.is_match(&self.content) {
+            for mat in re.find_iter(&self.content) {
+                trail.push_str(&self.content[mat.0..mat.1]);
+                trail.push('\n');
+            }
```

Comment:
That would be more elegant: in fact, it might not require an added function if we used join.